vscode/src/testController.ts

```diff
@@ -626,6 +628,50 @@ export class TestController {
     linkedCancellationSource.dispose();
   }
 
+  // When trying to a test file or directory, we need to know which framework is used by tests inside of it to resolve
+  // the command correctly. This method will resolve the first test file with children inside to determine the framework
+  // and then set that to all parents
+  private async discoverFrameworkTag(items: vscode.TestItem[]) {
+    const missingFramework = items.filter((item) => {
+      return !item.tags.some((tag) => tag.id.startsWith("framework"));
+    });
+
+    if (missingFramework.length === 0) {
+      return;
+    }
+
+    for (const item of missingFramework) {
+      let testFileItem = item;
+
+      while (!testFileItem.tags.some((tag) => tag === TEST_FILE_TAG)) {
+        const children: vscode.TestItem[] = [];
+
+        testFileItem.children.forEach((child) => {
+          children.push(child);
+        });
+
+        testFileItem = children[0];
+      }
+
+      // Handle the case where the test file is empty and has no children
+      await this.resolveHandler(testFileItem);
+      this.recursivelySetFrameworkTag(testFileItem);
+    }
+  }
+
+  private recursivelySetFrameworkTag(item: vscode.TestItem) {
+    const tag = item.tags.find((tag) => tag.id.startsWith("framework"))!;
+    let parent = item.parent;
+
+    while (parent) {
+      if (!parent.tags.some((tag) => tag.id.startsWith("framework"))) {
+        parent.tags = [...parent.tags, tag];
```

Comment:
Do we need a new array here?

```suggestion
        parent.tags.push(tag);
```